### ISSUE TITLE
Exporting macro and importing is not 1:1

### ISSUE THREAD

**Issue by Spidapida** [absolllute/Mega-Hack-Pro-Future#1125]
More specifically, exporting causes some actions to be wrong -- the `.mhr` works fine, but exporting then importing breaks the macro. For anyone who wants to test the macro, play this at ID 59626284, and add clicks at frame 24524 and 24578 (since what I'm doing here is to fix the macro using [this](https://matcool.github.io/gd-macro-converter/frame.html)). Also strangely, attempting to fix the macro in-game doesn't work -- I think what happened here is that MH didn't deleted the part I want to be deleted, causing the existing part to be played (but not the new one), but importing it again would coalesce all separated frame parts causing ghost and swift clicks to happen.

![image](https://user-images.githubusercontent.com/115771797/203189919-5c3eca7e-4ac7-4f4b-9f09-a55a4da02715.png)

File here: [macros.zip](https://github.com/absolllute/Mega-Hack-Pro-Future/files/10061680/macros.zip)

**Comment by NewAlexGamer** [comment]
exporting is not 1:1 because what you see in the json is the actual frame count, what you see on gd is an "event" count
![image](https://user-images.githubusercontent.com/86263500/203327381-45396a6c-61dc-46f1-9a9e-08b00fc5b849.png)
on the json i see 17961, while here you can clearly see a bigger number, that's because it's not a frame counter but an event counter. quite unusual and confusing i know

**Comment by Spidapida** [comment]
> exporting is not 1:1 because what you see in the json is the actual frame count, what you see on gd is an "event" count ![image](https://user-images.githubusercontent.com/86263500/203327381-45396a6c-61dc-46f1-9a9e-08b00fc5b849.png) on the json i see 17961, while here you can clearly see a bigger number, that's because it's not a frame counter but an event counter. quite unusual and confusing i know. the event thing was stated here [#955 (comment)](https://github.com/absolllute/Mega-Hack-Pro-Future/issues/955#issuecomment-1223067766) and if you're reffering to the fact that if you import the json back to mhr it breaks the macro, that's new actually

What? What I only showed on the screenshot I posted is that after the frame 25816 (which that and previous click events up to somewhere near frame 24524), the macro goes back to frame 24524, which doesn't make sense, for the most part anyway. Anyone is free to grab the macro on the `.zip` file I posted then exporting it for themselves in case you thought I simply added it post edit. The post pertains to the original macro and exported-imported macro not playing the same.

**Comment by Spidapida** [comment]
*reopened because dumb*

**Comment by NewAlexGamer** [comment]
ops, missread the issue sorry, were you using frame stepper by any chance?

**Comment by Spidapida** [comment]
> ops, missread the issue sorry, were you using frame stepper by any chance?

Yes, but also disabled retreat frame because right now, it deletes parts of macro if you retreat frame/s back to the frame before your latest checkpoint, which is incredibly annoying.

**Comment by Brittank88** [comment]
This is very strange - I've created multiple entire macros with frame stepper (including excessive use of the frame retreat) without issue.
